File: fixtures/coding/maintainability-mini-service-go/audit.go
package miniservice

type AuditEvent struct {
	Type    string
	ActorID string
	Payload map[string]any
}

// AuditLog is a tiny in-memory audit log used by tests and metrics fixtures.
type AuditLog struct {
	events []AuditEvent
}

func NewAuditLog() *AuditLog {
	return &AuditLog{}
}

func (a *AuditLog) Record(eventType string, actorID string, payload map[string]any) {
	a.events = append(a.events, AuditEvent{Type: eventType, ActorID: actorID, Payload: cloneMap(payload)})
}

func (a *AuditLog) ListEvents() []AuditEvent {
	events := make([]AuditEvent, len(a.events))
	for index, event := range a.events {
		events[index] = AuditEvent{Type: event.Type, ActorID: event.ActorID, Payload: cloneMap(event.Payload)}
	}
	return events
}

func cloneMap(input map[string]any) map[string]any {
	output := map[string]any{}
	for key, value := range input {
		output[key] = value
	}
	return output
}

File: fixtures/coding/maintainability-mini-service-rust/src/audit.rs
use crate::models::JsonMap;

#[derive(Clone, Debug, PartialEq)]
pub struct AuditEvent {
    pub event_type: String,
    pub actor_id: String,
    pub payload: JsonMap,
}

/// Tiny in-memory audit log used by tests and metrics fixtures.
#[derive(Clone, Debug, Default)]
pub struct AuditLog {
    events: Vec<AuditEvent>,
}

impl AuditLog {
    pub fn new() -> Self {
        Self { events: Vec::new() }
    }

    pub fn record(&mut self, event_type: &str, actor_id: &str, payload: JsonMap) {
        self.events.push(AuditEvent {
            event_type: event_type.to_string(),
            actor_id: actor_id.to_string(),
            payload,
        });
    }

    pub fn list_events(&self) -> Vec<AuditEvent> {
        self.events.clone()
    }
}
